generators/client/templates/react/src/main/webapp/app/shared/layout/header/header.tsx.ejs

```diff
@@ -76,10 +76,18 @@ export class Header extends React.Component<IHeaderProps, IHeaderState> {
   state: IHeaderState = {
       menuOpen: false
   };
+  <%_ if (enableI18nRTL) { _%>
+  componentDidMount () {
+    document.querySelector('html').setAttribute('dir', isRTL(Storage.local.get('locale')) ? 'rtl' : 'ltr');
+  };
+  <%_ } _%>
 
   <%_ if (enableTranslation) { _%>
   handleLocaleChange = event => {
     this.props.onLocaleChange(event.target.value);
+    <%_ if (enableI18nRTL) { _%>
+    document.querySelector('html').setAttribute('dir', isRTL(event.target.value) ? 'rtl' : 'ltr');
```

Comment:
I don't know if using a `setAttribute` was a good idea nor if it's the good place to put it.